Singer.API/ClientApp/src/app/modules/dashboard/components/event-list/event-list.component.ts

```diff
@@ -1,43 +1,50 @@
-import {
-   Component,
-   OnInit,
-   Input,
-   ViewChild,
-   Output,
-   EventEmitter,
-} from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import { EventDescription } from 'src/app/modules/core/models/singerevent.model';
-import { MatDrawer } from '@angular/material';
 import { SearchEventData } from '../event-search/event-search.component';
 import { SingerEventLocation } from 'src/app/modules/core/models/singer-event-location';
+import { SingerEventsService } from 'src/app/modules/core/services/singerevents-api/singerevents.service';
+import { SingerEventLocationService } from 'src/app/modules/core/services/singerevents-api/singerevent-location.service';
 
 @Component({
    selector: 'app-event-list',
    templateUrl: './event-list.component.html',
    styleUrls: ['./event-list.component.css'],
 })
 export class EventListComponent implements OnInit {
-   @Input() events: EventDescription[];
-   @Input() availableLocations: SingerEventLocation[];
+   breakpoint: number;
 
-   @Output() searchEvent: EventEmitter<SearchEventData> = new EventEmitter();
+   events: EventDescription[] = [];
+   availableLocations: SingerEventLocation[];
 
-   @ViewChild('drawer') drawer: MatDrawer;
-   breakpoint: number;
+   constructor(
+      private _eventService: SingerEventsService,
+      private _eventLocationService: SingerEventLocationService
+   ) {}
+
+   ngOnInit(): void {
+      this._eventLocationService
+         .fetchSingerEventLocationsData('asc', 'name', 0, 1000, '')
+         .subscribe(res => {
+            this.availableLocations = res.items as SingerEventLocation[];
+         });
 
-   constructor() {}
+      // Make first searchevent to load all events
+      var emptySearchEventData: SearchEventData = {
+         startDateTime: null,
+         endDateTime: null,
+         locationId: '',
+      };
 
-   ngOnInit() {
-      this.breakpoint = window.innerWidth <= 500 ? 1 : 3;
-      const searchEvent = <SearchEventData>{};
-      this.searchEvent.emit(searchEvent);
+      this.onSearchEvent(emptySearchEventData);
    }
 
    onResize(event) {
       this.breakpoint = event.target.innerWidth <= 500 ? 1 : 3;
```

Comment:
You could be able to get the current window in TS too.

Reply:
Yes this was causing that giant cards until resized bug. Fixed it. Thx!